### ISSUE TITLE
Remember me functionality?

### ISSUE THREAD

**Issue by bravegag** [joscha/play-authenticate#360]
Hi, is the remember me functionality implemented somewhere? maybe an open PR? I'd like to have that. I'm soon implementing. Also is there anyone who knows Scala well and can help me with this remember me functionality plus implementing the real thing i.e. securing the authentication with salt etc and I will be happy to hire for this over Upwork.

I'd like to have the improvements done in this fork:
https://github.com/bravegag/play-authenticate-usage-scala

and I am also happy to get them in the PA repo.

**Comment by joscha** [comment]
> I believe we'd need to use Deadbolt2 to add this sensitiveness or sudo category to certain Actions.

You might want to look at #345 first, to update to latest Play 2.6 before adding a bigger feature on top.

**Comment by bravegag** [comment]
Hi @joscha! thank you for referring me to #345 it really makes sense indeed to do that first. I know JPA well and can help switch to that I just don’t have a lot of time but could supervise it if needed.  However, AFAIK in PA the EBean bit is only used for the usage samples or? Another possibility is to split PA core from the usage sample that uses EBean, and just migrate the core. For example, in my Scala usage sample https://github.com/bravegag/play-authenticate-usage-scala I use Slick that is modern, faster and more productive than the alternatives furthermore, there I use a clean DAO pattern on top. You could use discontinue the Java usage sample until further notice and introduce the Scala one I have with Slick.  

One question, would anyone be willing to make a money pool to sponsor this work? I can put down $100 to get it done via Upwork. I really need time to market. I don't know but for me it is critical to have this working ... and if a few guys contribute $100 or less we could sponsor someone migrating this in parallel .. 

I will replicate this comment in the discussion over there #345 for convenience.

**Comment by bravegag** [comment]
Hi @leccyril, I understand your point and your concerns, if I had my applications with EBean I would feel the same. I have never used EBean in my projects and I was just giving alternatives to move PA forward. 

EBean is not part of the PA core which is what most people actually depend on. EBean is simply a technology choice for the samples. This means people who chose EBean should find the way to migrate their own applications ... but this is my 5c to the problem. I know it is not easy,

**Comment by bravegag** [comment]
I have moved my http://perfectjpattern.sourceforge.net/ project to Github and now with Apache license.
https://github.com/bravegag/perfectjpattern check out under jee DAO and JPA for the examples on different vendors.

**Comment by bravegag** [comment]
We are going to have a "Remember Me" PR soon. 

I propose to migrate to JPA via perfectjpattern it's very easy using the DAO interface ... @joscha you ok with this? We can choose any provider: OpenJPA, Hibernate, ElicpseLink .. which one?

**Comment by bravegag** [comment]
We have done some progress and here is a merge preview that includes the "Remember Me" functionality described before. The points 1) 2) 3) described above.

https://github.com/joscha/play-authenticate/compare/master...bravegag:master

**Comment by joscha** [comment]
Just had a short skim - gist is looking good. Let me know when it is ready.

**Comment by bravegag** [comment]
Hi @joscha! Could you please take a look at this and help with the momentum we currently have :) as soon as this is done and goes through I will look into the JPA bit to help PA move on to Play 2.6. I need this one to go through first. I think the only error we have is the CI looking and picking a different PA artifact. Locally all compiles and work perfectly.

**Comment by joscha** [comment]
Hi @bravegag - will have a look on Sunday night, sorry completely booked before. Keep the momentum, it is great!

**Comment by bravegag** [comment]
@joscha Any progress on this?

**Comment by joscha** [comment]
sorry, I think I had a look before the 2.6 upgrade - is it rebased against that? Also: can we decouple "remember me" from any change in the persistence layer?
